api/src/test/java/org/bahmni/module/communication/service/impl/MessageBuilderImplTest.java

```diff
@@ -0,0 +1,163 @@
+        String result = messageBuilderService.generateAppointmentMessage(arguments, providers, smsTemplate);
+
+        assertEquals("Generated Appointment SMS TemplateProvider TemplateTeleconsultation Link TemplateHelpdesk Template", result);
+    }
+
+    @Test
+    public void testGenerateAppointmentMessageWhenProviderIsEmpty() {
+        Map<String, String> arguments = new HashMap<>();
+        arguments.put("key", "value");
+        arguments.put("appointmentKind", "Virtual");
+        List<String> providers = new ArrayList<>();
+        String smsTemplate = "Appointment SMS Template";
+        String expectedTemplate = "Generated Appointment SMS Template";
+
+        when(smsTemplateService.templateMessage(smsTemplate, arguments)).thenReturn(expectedTemplate);
+        when(smsTemplateService.templateMessage(MessageBuilderServiceImpl.APPOINTMENT_TELECONSULTATION_LINK_TEMPLATE, arguments))
+                .thenReturn("Teleconsultation Link Template");
+        when(smsTemplateService.templateMessage(MessageBuilderServiceImpl.HELPDESK_TEMPLATE, arguments))
+                .thenReturn("Helpdesk Template");
+
+        String result = messageBuilderService.generateAppointmentMessage(arguments, providers, smsTemplate);
+
+        assertEquals("Generated Appointment SMS TemplateTeleconsultation Link TemplateHelpdesk Template", result);
+    }
+
+    @Test
+    public void testGenerateAppointmentMessageWhenAppointmentKindIsNotVirtual() {
+        Map<String, String> arguments = new HashMap<>();
+        arguments.put("key", "value");
+        arguments.put("appointmentKind", "InPerson");
+        List<String> providers = Collections.singletonList("Provider1");
+        String smsTemplate = "Appointment SMS Template";
+        String expectedTemplate = "Generated Appointment SMS Template";
+
+        when(smsTemplateService.templateMessage(smsTemplate, arguments)).thenReturn(expectedTemplate);
+        when(smsTemplateService.templateMessage(MessageBuilderServiceImpl.APPOINTMENT_PROVIDER_TEMPLATE, arguments))
+                .thenReturn("Provider Template");
+        when(smsTemplateService.templateMessage(MessageBuilderServiceImpl.HELPDESK_TEMPLATE, arguments))
+                .thenReturn("Helpdesk Template");
+
+        String result = messageBuilderService.generateAppointmentMessage(arguments, providers, smsTemplate);
+
+        assertEquals("Generated Appointment SMS TemplateProvider TemplateHelpdesk Template", result);
+    }
+
+
+    @Test
+    public void testGeneratePatientMessage() {
+        Map<String, String> arguments = Collections.singletonMap("key", "value");
+        String smsTemplate = "Patient SMS Template";
+        String expectedTemplate = "Generated Patient SMS Template";
+
+        when(smsTemplateService.templateMessage(smsTemplate, arguments)).thenReturn(expectedTemplate);
+
+        String result = messageBuilderService.generatePatientMessage(arguments, smsTemplate);
+
+        assertEquals(expectedTemplate, result);
```

Comment:
The input template says: "Patient SMS Templated" and expected says: "Generated Patient SMS Template". Where does the word "Generated" come in?

Reply:
Mock's return value is set to the string which includes Generated in it. I'll refactor the test to avoid confusion.